### PR TITLE
fix(aws-resources-functions): Save function configuration instead of failing on `AccessDenied` or `AccessDeniedException` errors in `aws_lambda_functions`

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

When `PreResourceResolver` returns with an error, the resolution for the rest of the resource is not even attempted...

### CODE REVIEW

File: plugins/source/aws/resources/services/lambda/functions_fetch.go [hunk 1]
```diff
@@ -13,21 +12,15 @@ import (
...
-	for {
-		response, err := svc.ListFunctions(ctx, &input)
+	svc := meta.(*client.Client).Services().Lambda
+	paginator := lambda.NewListFunctionsPaginator(svc, &lambda.ListFunctionsInput{})
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx)
 		if err != nil {
 			return err
 		}
+		res <- page.Functions
```
Comment (bbernays) [inline plugins/source/aws/resources/services/lambda/functions_fetch.go:23]: This is just cleanup

File: plugins/source/aws/resources/services/lambda/functions_fetch.go [hunk 2]
```diff
@@ -53,7 +46,10 @@ func getFunction(ctx context.Context, meta schema.ClientMeta, resource *schema.R
 			resource.Item = &lambda.GetFunctionOutput{
 				Configuration: &f,
 			}
+			c.Logger().Warn().Err(err).Msg("configuration data retrieved from ListFunctions will still be persisted")
+			return nil
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply plugins/source/aws/resources/services/lambda/functions_fetch.go:49]: To clarify, this is the only behavioral change? The other changes are refactoring?

**Review by erezrokah — APPROVED** [review]
Looks good, but maybe we should consider doing this in the SDK (i.e. moving to the next resource instead of failing)

**Comment by erezrokah** [comment]
I also updated the PR title so we can a more descriptive change log message

File: plugins/source/aws/resources/services/lambda/functions_fetch.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/resources/services/lambda/functions_fetch.go:49]: That is correct

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**